lectures/delta-time/README.md

```diff
@@ -0,0 +1,360 @@
+```
+
+<sub>**Figure 8**: Our character facing an item.</sub>
+
+If we were to translate both items _with respect to the world_, the code would be similar to what we've seen before:
+
+```c++
+character_model_matrix = glm::mat4(1.0f);
+character_model_matrix = glm::translate(character_model_matrix, glm::vec3(TRANS_VALUE, TRANS_VALUE, 0.0f));
+
+item_model_matrix = glm::mat4(1.0f);
+item_model_matrix = glm::translate(item_model_matrix, glm::vec3(TRANS_VALUE, TRANS_VALUE, 0.0f));
+```
+
+<sub>**Code Block 1**: Translating both items in the world space.</sub>
+
+What happens when our character picks up the item, though? The item ceases to follow its own "space", and adopts the character's:
+
+```
+                   +
+                  /
+                 /
+            +---+
+            |   |
+            +---+
+—————————————————————————————————————
+      Character + Item
+```
+
+<sub>**Figure 9**: Our character is now holding the item, and carrying it everywhere with them.</sub>
+
+Our code now has to account for both the character's transformations **and** for the item's own transformations:
+
+```c++
+character_model_matrix = glm::mat4(1.0f);
+character_model_matrix = glm::translate(character_model_matrix, glm::vec3(TRANS_VALUE, TRANS_VALUE, 0.0f));
+
+item_model_matrix = glm::translate(character_model, glm::vec3(TRANS_VALUE, TRANS_VALUE, 0.0f));
+item_model_matrix = glm::rotate(character_model_matrix, ANGLE, glm::vec3(0.0f, 0.0f, 1.0f));
+```
+
+<sub>**Code Block 2**: The character is probably still translated with respect to the world space, but our item now also has to be translated with respect to our character; in a sense, the character's space becomes the item's world space.</sub>
+
+---
+
+Now, it goes without saying that games require movement to actually be games. What we have been doing so far is something akin to the following:
+
+```c++
+glm::mat4 model_matrix;
+
+void initialise() 
+{
+    model_matrix = glm::mat4(1.0f);
+}
+
+void update()
+{
+    model_matrix = glm::translate(model_matrix, glm::vec3(0.1f, 0.0f, 0.0f));
+}
+```
+
+<sub>**Code Block 3**: Your basic game look animation.</sub>
+
+This is how games have been programmed and thought of since the dawn of the industry—but it is now, by and large, antiquated and can cause weird things to happen.
+
+Take as an example something that happened to several people in the class: their triangle was either spinning slower than mine, or oftentimes much faster than mine. This is not uncommon—after all, mine is a laptop from 2022 that is in no way set up to provide a crazy framerate. This is a problem, though, because when we make games, we want them all to run at the same "speed," regardless of the machine that it is running on. For this reason, we will change two things in our code. 
+
+The first is, instead of keeping track of the matrix's transformations per frame, we keep track of the arguments values being passed into them:
+
+```c++
+float model_x;
```

Comment:
Please add `g_` here as well, as well in all further uses of these variables below.